api/services/Flairs.js

```diff
@@ -8,6 +8,44 @@ var alolaFlair = ['rowlet', 'litten', 'popplio'];
 var eventFlair = kantoFlair.concat(alolaFlair);
 var eventFlairRegExp = new RegExp('\\bkva-(' + eventFlair.join("|") + ')-[1-3]\\b');
 
+// Mappings from css_flair to emoji names
+const ptrades_mappings = {
+  "default" : ":0:",
+  "gen2" : ":2:",
+  "pokeball" : ":10:",
+  "premierball" : ":20:",
+  "greatball" : ":30:",
+  "ultraball" : ":40:",
+  "luxuryball" : ":50:",
+  "masterball" : ":60:",
+  "dreamball" : ":70:",
+  "cherishball" : ":80:",
+  "ovalcharm" : ":90:",
+  "shinycharm" : ":100:",
+  "gsball" : ":GS:"
+};    
+
+const svex_egg_mappings = {
+  "lucky" : ":1e:",
+  "egg" : ":5e:",
+  "eevee" : ":10e:",
+  "togepi" : ":20e:",
+  "torchic" : ":30e:",
+  "pichu" : ":50e:",
+  "manaphy" : ":75e:",
+  "eggcup" : ":100e:"
+};
+
+const svex_ribbon_mappings = {
+  "cuteribbon" : "30r",
+  "coolribbon" : "100r",
+  "beautyribbon" : "200r",
+  "smartribbon" : "400r",
+  "toughribbon" : "800r"
```

Comment:
These emoji mappings seem like they could be a pain to modify if the requirements for flair ever change (e.g. if you decide that egg cup should require 120 hatches instead of 100, there will still be a bunch of users with `:100e:` in their flair text). Have you considered naming the emojis with the an abbreviation for the emoji itself instead? For example, `eggcup` could map to something like `:ec:` instead of `:100e:`.

Reply:
Changed the mappings into tiers for svex. Since we're requiring 10 trades (currently pokeball) to trade events and shinies, we're gonna keep the mappings for ptrades though!